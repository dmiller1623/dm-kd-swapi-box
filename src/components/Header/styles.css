.header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 90vw; }

.favorites-button {
  height: 3rem;
  width: 8rem;
  display: flex;
  align-items: center;
  justify-content: space-between; }
