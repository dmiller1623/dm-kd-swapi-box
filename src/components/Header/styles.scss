.header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 90%;
}

.favorites-button {
  height: 3rem;
  width: 10rem;
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.fav-num {
  margin-left: 10px;
}

.favorites-div {
  cursor: pointer;
}

img {
  margin-left: 9rem;
}