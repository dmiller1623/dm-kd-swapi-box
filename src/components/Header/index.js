import React from 'react';
import './styles.css'
import PropTypes from 'prop-types'

const Header = ({ numOfFaves, handleDisplayCards, evaluateButtonClass }) => {
  return (
    <div className='header'>
      <div></div>
      <img src='https://fontmeme.com/permalink/180823/038124195840fbb42b03e437645af736.png'></img>
      <div className='favorites-button'>
        <p className={evaluateButtonClass('favorites')} id='favorites' onClick={(e) => handleDisplayCards(e)}>View Favorites
        <div className='fav-num'>{numOfFaves}</div>
        </p>
      </div>
    </div>
  )
}

Header.propTypes = {
  numOfFaves: PropTypes.number,
  handleDisplayCards: PropTypes.func
}

export default Header;

// thanks to fontmeme.com for the header font link!