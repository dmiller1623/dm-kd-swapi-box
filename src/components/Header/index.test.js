import React from 'react'
import Header from './index.js'
import { shallow } from 'enzyme'

describe('Header', () => {
  let wrapper 
  let mockHandleDisplay

  beforeEach(() => {
    mockHandleDisplay = jest.fn()
    wrapper = shallow(<Header handleDisplayCards={mockHandleDisplay}/>)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should call handleDisplay cards when the fav button is clicked', () => {
    wrapper.find('.favorites-div').simulate('click')
    expect(mockHandleDisplay).toHaveBeenCalled()
  })
})