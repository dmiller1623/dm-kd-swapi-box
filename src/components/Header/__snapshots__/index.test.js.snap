// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    handleDisplayCards={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div />,
        <h1>
          SWAPI-Box
</h1>,
        <div
          className="favorites-button"
>
          <p
                    className="favorites-div"
                    onClick={[Function]}
          >
                    View Favorites
          </p>
          <div />
</div>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "SWAPI-Box",
        },
        "ref": null,
        "rendered": "SWAPI-Box",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="favorites-div"
              onClick={[Function]}
>
              View Favorites
</p>,
            <div />,
          ],
          "className": "favorites-button",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "View Favorites",
              "className": "favorites-div",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "View Favorites",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div />,
          <h1>
            SWAPI-Box
</h1>,
          <div
            className="favorites-button"
>
            <p
                        className="favorites-div"
                        onClick={[Function]}
            >
                        View Favorites
            </p>
            <div />
</div>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "SWAPI-Box",
          },
          "ref": null,
          "rendered": "SWAPI-Box",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="favorites-div"
                onClick={[Function]}
>
                View Favorites
</p>,
              <div />,
            ],
            "className": "favorites-button",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "View Favorites",
                "className": "favorites-div",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "View Favorites",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
